cpp/include/cugraph_c/layout_algorithms.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <cugraph_c/error.h>
+#include <cugraph_c/graph.h>
+#include <cugraph_c/graph_functions.h>
+#include <cugraph_c/random.h>
+#include <cugraph_c/resource_handle.h>
+
+/** @defgroup layout Layout algorithms
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @brief   Force Atlas 2
+ *
+ * NOTE: This currently wraps the legacy force atlas 2 implementation and is only
+ * available in Single GPU implementation.
+ *
+ * @param [in]   handle          Handle for accessing resources
+ * @param [in]   graph           Pointer to graph.  NOTE: Graph might be modified if the storage
+ *                               needs to be transposed
+ * @param [out]  pos             Position of the vertices
+ * @param [in]   max_iter        Maximum number of iterations. Initial vertex positioning
+ * @param [in]   x_start         Initial vertex positioning (x-axis)
+ * @param [in]   y_start         Initial vertex positioning (y-axis)
+ * @param [in]   outbound_attraction_distribution
+ *                               Distributes attraction along outbound edges
+ *                               Hubs attract less and thus are pushed to the borders.
+ * @param [in]   lin_log_mode    Switch Force Atlas model from lin-lin to lin-log.
+ *                               Makes clusters more tight.
+ * @param [in]   prevent_overlapping
+ *                               Prevent nodes to overlap.
+ * @param [in]   edge_weight_influence
+ *                               How much influence you give to the edges weight.
+ *                               0 is “no influence” and 1 is “normal”.
+ * @param [in]   jitter_tolerance
+ *                               How much swinging you allow. Above 1 discouraged.
+ *                               Lower gives less speed and more precision.
+ * @param [in]   barnes_hut_optimize
+ *                               Whether to use the Barnes Hut approximation or the slower exact version.
+ * @param [in]   barnes_hut_theta
+ *                               Float between 0 and 1. Tradeoff for speed (1) vs accuracy (0) for Barnes Hut only.
+ * @param [in]   scaling_ratio
+ *                               How much repulsion you want. More makes a more sparse graph.
+ *                               Switching from regular mode to LinLog mode needs a readjustment of the scaling parameter.
+ * @param [in]   strong_gravity_mode
+ *                               Sets a force that attracts the nodes that are distant from the
+ *                               center more. It is so strong that it can sometimes dominate other forces.
+ * @param [in]   gravity         Attracts nodes to the center. Prevents islands from drifting away.
+ * @param [in]   verbose         Output convergence info at each interation.
+ * @param [in]   do_expensive_check
+ *                               A flag to run expensive checks for input arguments (if set to true)
+ * @param [out]  result          Opaque object containing the clustering result
```

Comment:
This result would be what `pos` above is.  I'd delete the `pos` parameter above and move it to this position.  By convention, the C API functions pass all input parameters before the output parameters.